exporters/otlp/partialsuccess.go

```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlp // import "go.opentelemetry.io/otel/exporters/otlp"
```

Comment:
I'm conflicted about where this should live.  On the one hand, I would like users to create an ErrorHandler that does something more intelligent than just log a string.  On the other hand, I would prefer not to expose a new API surface needlessly.

If no one has any other objections to this than I can live with it.

Reply:
I'm a fan of structured logging; I would advocate for a backwards-compatible mechanism to accomplish it. The `PartialSuccess` error can implement some future API that enables it to be expressed in a structured way when the producer logs it to a structure-aware logger.

I didn't know where to put it either. At this level in the current package structure, it falls into the top-level package, which feels OK to me.

Reply:
With no objections to this, we can leave it where it is.

Reply:
I would rather have these live in an internal package, at lease to start.

Reply:
I'm not sure what a user will be able to do with an error handler other than log the error given they cannot retry the export. And adding this to the public API seems like adding something only OTel developers will use or care about, not users.